pkg/cmd/server/start.go

```diff
@@ -70,16 +70,36 @@ type PostStartHook struct {
 	Name string
 }
 
+type StartOptions struct {
+	EtcdPath         string
+	Apis             []*builders.APIGroupBuilder
+	Openapidefs      openapi.GetOpenAPIDefinitions
+	Title            string
+	Version          string
+	TweakConfigFuncs []func(apiServer *apiserver.Config) error
+
+	//FlagConfigFunc handles user-defined flags
+	FlagConfigFuncs []func(*cobra.Command) error
+}
+
 // StartApiServer starts an apiserver hosting the provider apis and openapi definitions.
-func StartApiServer(etcdPath string, apis []*builders.APIGroupBuilder, openapidefs openapi.GetOpenAPIDefinitions, title, version string, tweakConfigFuncs ...func(apiServer *apiserver.Config) error) {
```

Comment:
this is breaking backward-compatibility. how about start w/ a new function like `StartApiServerWithOptions` and preserve the old one w/ a deprecation note. and the old one should delegate to the new like:

```go
func StartApiServer(etcdPath string, apis []*builders.APIGroupBuilder, openapidefs openapi.GetOpenAPIDefinitions, title, version string, tweakConfigFuncs ...func(apiServer *apiserver.Config) error) {
  _ = StartApiServerWithOptions(&StartOptions{...})
}
```